### ISSUE TITLE
gss_add_cred() may unintentionally duplicate a mechanism cred

### ISSUE THREAD

**Issue by greghudson** [heimdal/heimdal#414]
If an input_cred_handle is given, mechglue gss_add_cred() loops over its mechanism creds with the comment:
```
        /*                                                                      
         * We go through all the mc attached to the input_cred_handle           
         * and check the mechanism. If it matches, we call                      
         * gss_add_cred for that mechanism, otherwise we copy the mc            
         * to new_cred.                                                         
         */
```
The loop sets target_mc if it finds a matching mechanism cred, and later uses target_mc when calling the mechanism's gss_add_cred() function.  I do not really understand the use case for this logic, but the intent is clearly that the newly created credential should have only one mechanism cred for desired_mech.  However, there is no `break` or `else` when the loop sets target_mc, so it will copy the pre-add_cred mechanism credential into the new cred; the function will then add the post-add_cred mechanism credential at the end, for a total of two mechanism credentials for desired_mech.

**Comment by lhoward** [comment]
Should be fixed by e6d1c108, @nicowilliams?

**Comment by nicowilliams** [comment]
Yes, this is fixed by e61c10, indeed.  We can now have only one element per-mechanism, though each element can have more than one element itself.
